spec/factories/billing_entities.rb

```diff
@@ -2,14 +2,18 @@
 
 FactoryBot.define do
   factory :billing_entity do
-    organization
     name { Faker::Company.name }
     code { "entity_#{SecureRandom.uuid}" }
     default_currency { "USD" }
 
     email { Faker::Internet.email }
     email_settings { ["invoice.finalized", "credit_note.created"] }
 
+    after :build do |billing_entity, values|
+      billing_entity.is_default = true if values.organization&.billing_entities&.where(is_default: true).blank?
```

Comment:
Good point.

However for a test case I would not rely on setting `is_default` automatically and use the trait to specifically do so. Otherwise we could run into unexpected/hidden behaviours because the `is_default` is set to true and we don not expect it... also this will ignore the fact of archived entities, etc.

Reply:
my idea was that since most of the cases in production and test-cases as well per each organization we'll have only one billing_entity and for the correct flow it should be created, it will make test set-up easier to always create an organization with default billing entity. so when we're creating a billing entity, if it's the only entity of an organization, it should be default....

However if we're creating explicitly exactly billing_entity, it makes sense that the person who's writing the test is aware of billing_entities architecture...  and might choose is_default/no themselves...

yep, I'll remove the magic from here

Reply:
discussed it offline, left a comment that it's needed temporary and will need to be deleted after the final implementation complete